Add pictures and the text for installation on Raspberry Pi (#132)
opencloud-eu/docs: docs/admin/20-getting-started/50-raspberry-pi.md

sidebar_position: 5
id: raspberry-pi
title: Raspberry Pi
---

# OpenCloud on Raspberry Pi

## 1.1 Hardware requirements

> - Minimum Raspberry Pi 4B with at least 4 GB RAM connected via LAN or WLAN

Reviewer: Sometimes the wording is Raspberry Pi, sometimes Raspberry-Pi

- Micro SD card with at least 32 GB storage space
- External hard disk or USB-Stick (optional) for additional storage space

## 1.2 Install operating system

> - Install OS  

Reviewer: ```suggestion
- Install Raspberry Pi OS
```

## 1.3 Connecting with SSH

Start the Raspberry Pi with the SD card and connect via SSH.  
The IP for this can be viewed in your router.

<img src={require("./img/raspberrypi/ip-router.png").default} alt="find ip from raspberry-pi in router" width="500"/>

#### Establish connection via SSH:
```sh
ssh pi@YOUR-IP
```

After the first login, you should change the password for security reasons:
```sh
passwd
```

## 1.4 Installing Docker and Docker Compose

Detailed installation instructions for docker can be found here:  
[Install Docker on Raspberry Pi](https://pimylifeup.com/raspberry-pi-docker/)

- Perform update and upgrade

```sh
sudo apt update && sudo apt upgrade -y
```
- Install Docker via Script

```sh
curl -fsSL test.docker.com -o get-docker.sh && sh get-docker.sh
```

- Add current user to the Docker Group

```sh
sudo usermod -aG docker ${USER}
```
- Check if it`s running:

```sh
groups ${USER}
```
<img src={require("./img/raspberrypi/docker-user-check.png").default} alt="Check docker user" width="500"/>

- Reboot the Raspberry Pi to let the changes take effect

```sh
sudo shutdown -r now
```
- Install Docker-Compose

```sh

> sudo apt install docker-compose

Reviewer: Do we need to install docker at all when we install docker compose ?

Author: yes

```

- Check the installation

```sh

> docker-compose --version

Reviewer: docker-compose vs docker compose in the other commands

Author: you really need sometime `docker-compose` but I will change all to `docker-compose`

```
<img src={require("./img/raspberrypi/docker-compose-check.png").default} alt="Check docker-compose" width="500"/>

## 1.5 Clone OpenCloud respository

```sh 
git clone https://github.com/opencloud-eu/opencloud.git
``` 

## 1.6 Start the docker compose setup

```sh 
cd opencloud/deployments/examples/opencloud_full
```

```sh 
docker compose up 
```

Now you have running OpenCloud locally on your RaspberryPi and you can adjust it to your needs.

> We will describe how you can mount an external disk or USB-Stick and make your OpenCloud available from outside with noip

Reviewer: ```suggestion
We will describe how you can mount an external disk or USB-Stick and make your OpenCloud available outside of the local network with No-IP
```

```sh
sudo mkfs.ext4 PATH-TO-DRIVE -L DATA
```
PATH-TO-DRIVE is in this example `/dev/sda1` so the command would be

```sh
sudo mkfs.ext4 /dev/sda1 -L DATA
```
<img src={require("./img/raspberrypi/format-drive.png").default} alt="format drive" width="500"/>

#### 3. Add entry in fstab for automatic mounting when restarting

  - open fstab with sudo
```sh
  sudo nano /etc/fstab
```
  - add following line in the fstab file

```sh 
LABEL=DATA /mnt/data ext4 auto,defaults 0 0
```

#### 4. Create the `/mnt/data` directory and give the user 1000 access

```sh 
sudo mkdir -p /mnt/data
```
```sh 
sudo chown -R 1000:1000 /mnt/data
```

#### 5.  mount the drive automatically 

```sh 
sudo mount -a
```  

Maybe you get following error

<img src={require("./img/raspberrypi/error-mounting.png").default} alt="error mounting" width="500"/>

then please perform the reccomended command 

```sh 
systemctl daemon-reload
``` 
and try to mount again.

## 1.8 Mount external storage

Stop the running docker

```sh 
docker compose down
``` 

Go to the deployment folder and open the `.env` file with e.g. nano 

```sh 
cd opencloud/deployments/examples/opencloud_full
``` 
```sh 
nano .env
```

> When you added an external hard disk or USB-Stick for the storage, you need to enable the `OC_DATA_DIR` variable and change the path to your storage

Reviewer: ```suggestion
When you added an external hard disk or USB-Stick for the storage, you need to set the `OC_DATA_DIR` variable and adjust the path to your storage
```

<img src={require("./img/raspberrypi/noip.png").default} alt="noip hostname input" width="500"/>

#### 2. Configure DyDNS in your router

If your router supports integrated Dynamic DNS (DDNS), you can update your IP address directly via the router. This eliminates the need to install the Dynamic Update Client (DUC) from No-IP on your Raspberry Pi.
How to set up DDNS in the router:

- Log into your router's web interface (usually via 192.168.1.1 or a similar address).
- Search for the DDNS settings - these are usually located under Network, Internet or Dynamic DNS.
- Select No-IP as your provider from the list of supported DDNS services.
- Enter your No-IP credentials (username & password).
- Enter the hostname you previously created with No-IP (e.g. ocis-at-home.ddns.net).
- Save the settings and test the connection.

The router will now automatically update your public IP address at No-IP so that your Raspberry Pi always remains accessible under the selected subdomain.

You can also look in the documentation from [No-IP](https://www.noip.com/support/knowledgebase/how-to-configure-ddns-in-router)

> #### 3. Configure portforwarding in your router

Reviewer 2: ```suggestion
#### 3. Configure port forwarding in your router
```
